tests: frequent test failures during cover step

## What happened?
Our tests frequently fail during the cover step. Maybe 50% of the time? And it seems always in the same place:

https://github.com/pomerium/pomerium/runs/3225129332?check_suite_focus=true

```
{"level":"info","service":"redis","time":"2021-08-02T21:37:42Z","message":"redis: discarding bad PubSub connection: read tcp 172.17.0.1:58086->172.17.0.2:6379: use of closed network connection"}
{"level":"info","service":"redis","time":"2021-08-02T21:37:43Z","message":"redis: discarding bad PubSub connection: read tcp [::1]:59278->[::1]:49176: use of closed network connection"}
{"level":"info","service":"redis","time":"2021-08-02T21:37:53Z","message":"redis: discarding bad PubSub connection: read tcp [::1]:59286->[::1]:49176: use of closed network connection"}
--- FAIL: TestChangeSignal (10.22s)
    redis_test.go:179: 
        	Error Trace:	redis_test.go:179
        	            				redis.go:79
        	            				redis_test.go:132
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestChangeSignal
        	Messages:   	expected signal to be fired when another backend triggers a change
{"level":"info","service":"redis","time":"2021-08-02T21:37:55Z","message":"redis: discarding bad PubSub connection: read tcp [::1]:44242->[::1]:49177: use of closed network connection"}
{"level":"info","service":"redis","time":"2021-08-02T21:37:55Z","message":"redis: discarding bad PubSub connection: read tcp [::1]:45850->[::1]:49178: use of closed network connection"}
{"level":"info","service":"redis","time":"2021-08-02T21:37:56Z","message":"redis: discarding bad PubSub connection: read tcp [::1]:40074->[::1]:49179: use of closed network connection"}
```

## What did you expect to happen?
Not to fail

## How'd it happen?
1. Create a pull request
2. Test fails

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version`): master: https://github.com/pomerium/pomerium/commit/a64e5b5fa1a84fb257e4d6b24c878801585e6edc
- Server Operating System/Architecture/Cloud: github

--- a/pkg/storage/redis/redis_test.go
+++ b/pkg/storage/redis/redis_test.go
@@ -124,12 +124,10 @@ func TestChangeSignal(t *testing.T) {
 	}
 
 	ctx := context.Background()
-	ctx, clearTimeout := context.WithTimeout(ctx, time.Second*10)
-	defer clearTimeout()
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
 	require.NoError(t, testutil.WithTestRedis(false, func(rawURL string) error {
+		ctx, clearTimeout := context.WithTimeout(ctx, time.Second*30)
+		defer clearTimeout()
+
 		ready := make(chan struct{})
 		var eg errgroup.Group
 		eg.Go(func() error {

